File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.py
class Solution:

    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [0]* (len(cost)+1)
        i = 2
        dp[0] = 0
        dp[1] = 0
        while i <= len(cost):
            dp[i] = min(cost[i-1]+dp[i-1], cost[i-2]+dp[i-2])
            i+=1
        return dp[len(cost)]



        

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.cpp
class Solution {
public:
    int dp(int i, vector<int>&cost, vector<int>& memo){
        if(i>=cost.size()) return 0;
        if(memo[i] != -1) return memo[i];
        return memo[i] = cost[i] + min(dp(i+1, cost, memo),dp(i+2, cost, memo));
    }
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> memo(cost.size()+1, -1);
        return min(dp(0, cost, memo), dp(1, cost, memo));

    }
};